vec/vec.mbt

```diff
@@ -40,27 +40,58 @@ pub fn Vec::with_capacity[T](cap : Int) -> Vec[T] {
   Vec::{ buf: UninitializedArray::make(cap), len: 0 }
 }
 
-/// Adds an element to the end of the vector.
-pub fn push[T](self : Vec[T], value : T) {
-  if self.len == self.buf.length() {
-    let old_cap = self.len
-    let new_cap = if old_cap == 0 { 8 } else { old_cap * 2 }
-    let new_buf = UninitializedArray::make(new_cap)
-    for i = 0; i < old_cap; i = i + 1 {
-      new_buf[i] = self.buf[i]
-    }
-    new_buf[old_cap] = value
-    self.buf = new_buf
-    self.len = old_cap + 1
-  } else {
-    self.buf[self.len] = value
-    self.len += 1
+/// Creates a new vector from an array.
+pub fn Vec::from_array[T](arr : Array[T]) -> Vec[T] {
+  let len = arr.length()
+  let buf = UninitializedArray::make(len)
+  for i = 0; i < len; i = i + 1 {
+    buf[i] = arr[i]
+  }
+  Vec::{ buf, len }
+}
+
+/// Returns the number of elements in the vector.
+pub fn length[T](self : Vec[T]) -> Int {
+  self.len
+}
+
+/// Returns the total number of elements the vector can hold without reallocating.
+pub fn capacity[T](self : Vec[T]) -> Int {
+  self.buf.length()
+}
+
+/// Reallocate the vector with a new capacity.
+pub fn realloc[T](self : Vec[T]) {
```

Comment:
I think `realloc` should be private. Maybe we should provide `reserve` like in Rust std?

Reply:
You're right, I accidentally made it public. And `reserve` is a nice idea, I will add it later.